clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1579,6 +1579,108 @@ OpFoldResult cir::VecExtractOp::fold(FoldAdaptor adaptor) {
   return elements[index];
 }
 
+//===----------------------------------------------------------------------===//
+// VecCmpOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult cir::VecCmpOp::fold(FoldAdaptor adaptor) {
+  mlir::Attribute lhs = adaptor.getLhs();
+  mlir::Attribute rhs = adaptor.getRhs();
+  if (!mlir::isa_and_nonnull<cir::ConstVectorAttr>(lhs) ||
+      !mlir::isa_and_nonnull<cir::ConstVectorAttr>(rhs))
+    return {};
+
+  auto lhsVecAttr = mlir::cast<cir::ConstVectorAttr>(lhs);
+  auto rhsVecAttr = mlir::cast<cir::ConstVectorAttr>(rhs);
+
+  mlir::Type inputElemTy =
+      mlir::cast<cir::VectorType>(lhsVecAttr.getType()).getElementType();
+  if (!isAnyIntegerOrFloatingPointType(inputElemTy))
+    return {};
+
+  cir::CmpOpKind opKind = adaptor.getKind();
+  mlir::ArrayAttr lhsVecElhs = lhsVecAttr.getElts();
+  mlir::ArrayAttr rhsVecElhs = rhsVecAttr.getElts();
+  uint64_t vecSize = lhsVecElhs.size();
+
+  auto resultVecTy = mlir::cast<cir::VectorType>(getType());
+
+  SmallVector<mlir::Attribute, 16> elements(vecSize);
+  for (uint64_t i = 0; i < vecSize; i++) {
+    mlir::Attribute lhsAttr = lhsVecElhs[i];
+    mlir::Attribute rhsAttr = rhsVecElhs[i];
+
```

Comment:
is it possible to refactor to check `mlir::isa<cir::IntAttr>(lhsAttr)`  only once based on that obtain directly `lhsValue` and `rhsValue` and use those in subsequent cases?